File: src/main/java/casa_de_cha/model/Categoria.java
package casa_de_cha.model;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serial;
import java.io.Serializable;

@NoArgsConstructor
@AllArgsConstructor
@Entity
@SequenceGenerator(name = "seq_categoria", sequenceName = "seq_categoria", allocationSize = 1)
public class Categoria implements Serializable{

    @Serial
    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seq_categoria")
    private int id_categoria;
    @Column
    private String nome_cat;

    @Column
    private boolean ativo;

    public boolean isAtivo() {
        return ativo;
    }

    public void setAtivo(boolean ativo) {
        this.ativo = ativo;
    }

    public int getId_categoria() {
        return id_categoria;
    }

    public void setId_categoria(int id_categoria) {
        this.id_categoria = id_categoria;
    }

    public String getNome_cat() {
        return nome_cat;
    }

    public void setNome_cat(String nome) {
        this.nome_cat = nome;
    }

}

File: front-end/casa_de_cha/src/core/Categoria.ts
export default class Categoria {
    private _id: string
    private _nome: string
    private _ativo: boolean

    constructor(nome: string, ativo: boolean, id: string){
        this._nome = nome
        this._ativo = ativo
        this._id = id
    }

    static vazio() {
        return new Categoria('', true, '')
    }

    get id() {
        return this._id
    }

    get nome() {
        return this._nome
    }

    get ativo() {
        return this._ativo
    }
}
